HIP/hip-991.md

```diff
@@ -0,0 +1,221 @@
+* Cannot add Key after creation - similar to tokens.
+
+**Benefits:**
+
+* Provides predictable revenue per message for topic operators.
+* Enables new products and business models on HCS.
+* Increases node revenue for network fees tied to new HCS transactions/topic creations with fixed fees.
+* Enhances the utility of HCS within Hedera.
+* Increases revenue streams for node operators.
+* Enables simple facilitation of holding and spending tokens to document new information on a topic.
+
+**Related Requests:** This enhancement is in response to community feedback (Tier Bot) requesting more versatile and predictable economic models within topic IDs on the Hedera network.
+
+## **Specification**
+
+### **Conceptual Overview**
+
+We propose adding a fixed fee mechanism to the Hedera Consensus Service (HCS) for topic messages, similar to the custom fee structures in the Hedera Token Service (HTS). This will involve creating new Protobuf messages and modifying existing ones to accommodate fixed fees for topics.
+
+### **New Protobuf Messages**
+
+### **FixedFee**
+
+The `FixedFee` message represents a fixed fee charged for each message submission to a topic.
+
+```
+message FixedFee {
+  // The number of units to assess as a fee
+  int64 amount = 1;
+  // The denomination of the fee; taken as hbar if left unset
+  TokenID denominating_token_id = 2;
+}
+```
+
+### **CustomFee**
+
+The `CustomFee` message defines the type of fee and the account receiving the fee assessed during a message submission to the associated topic. A custom fee may only be fixed and must specify a fee collector account to receive the assessed fees.
+
+```
+message CustomFee {
+  oneof fee {
+    // Fixed fee to be charged
+    FixedFee fixed_fee = 1;
+  }
+  // The account to receive the custom fee
+  AccountID fee_collector_account_id = 3;
+}
+```
+
+### **Modified Protobuf Messages**
+
+### **TopicCreateTransactionBody**
+
+The `TopicCreateTransactionBody` message is updated to include an optional `custom_fees` property for specifying fixed fees during topic creation.
+
+```
+message TopicCreateTransactionBody {
+  // The ID of the topic to be created
+  TopicID topic_id = 1;
+  // The custom fees to be assessed during a message submission to this topic
+  repeated CustomFee custom_fees = 2;
+}
+```
+
+### **TopicInfo**
+
+The `TopicInfo` message is updated to include the current list of custom fixed fees associated with the topic.
+
+```
+message TopicInfo {
+  // The ID of the topic
+  TopicID topic_id = 1;
+  // The custom fees to be assessed during a message submission to this topic
+  repeated CustomFee custom_fees = 2;
+}
+```
+
+### **TransactionRecord**
+
+The `TransactionRecord` message is updated to include the list of custom fees assessed as part of executing the transaction represented by this record.
+
+```
+message TransactionRecord {
+  // All custom fees that were assessed during a message submission
+  repeated AssessedCustomFee assessed_custom_fees = 13;
```

Comment:
This message also exist. We should just update the documentation in protobufs.

Reply:
Added the note for this